shuffle.py

```diff
@@ -34,7 +32,7 @@ def get_shuffle_params(params, index):
     random.seed(0)
     keys = sorted(params.keys())
     iters = [iter_shuffle(params[key]) for key in keys]
-    vals = next(itertools.islice(six.moves.zip(*iters), index, None))
+    vals = next(itertools.islice(zip(*iters), index, None))
```

Comment:
`zip(*iters)` does not halt with Python 2 because `iters` are infinite.
```suggestion
    vals = tuple(iter[index] for iter in iters)
```